drivers/serial/uart_rpi_pico.c

```diff
@@ -90,6 +90,75 @@ static int uart_rpi_init(const struct device *dev)
 	return 0;
 }
 
+static int uart_rpi_config_get(const struct device *dev, struct uart_config *cfg)
+{
+	struct uart_rpi_data *data = dev->data;
+
+	*cfg = data->uart_config;
+	return 0;
+}
+
+static int uart_rpi_configure(const struct device *dev, const struct uart_config *cfg)
+{
+	const struct uart_rpi_config *config = dev->config;
+	uart_inst_t * const uart_inst = config->uart_dev;
+	struct uart_rpi_data *data = dev->data;
+
+	int baudrate = uart_set_baudrate(uart_inst, cfg->baudrate);\
```

Comment:
This should also validate all parameters before making any configuration changes, if one setting is not valid, no UART configuration change should occur